Add agenda section

haywirebanana opened the pull request:
I tried to implement a collapsable/accordion feature for the different dropdowns, but was unsure about how to deal with some HTMLAttribute errors so I decided to go for Detail & Summary Tags for ease of dealing with event handlers. 

![image](https://github.com/user-attachments/assets/fbf066a0-381a-4db0-a46f-0b12a4a94949)

NobleTyler committed:
Update README.md

NobleTyler committed:
Add pnpm to readme

NobleTyler committed:
Update README.md

lindan4 committed:
Created file.

lindan4 committed:
Rebased and updated file.

lindan4 committed:
Added styling for anchor elements, but still need to handle styling for last element.

lindan4 committed:
Added border and border radius.

lindan4 committed:
Added code to cloud icon as svg and to nav bar.

lindan4 committed:
Added column gutter and animation.

lindan4 committed:
Upgraded Astro.

victoriacheng15 committed:
add header component

victoriacheng15 committed:
add code to cloud logo and background for the header

haywirebanana committed:
detail+summary tags

haywirebanana committed:
detail+summary tags

haywirebanana committed:
general css done

haywirebanana committed:
general css done

haywirebanana committed:
fixed hover issue

haywirebanana committed:
fixed hover issue

lindan4 commented:
I can see that the package-lock.json file was modified, but it doesn't look like any changes were made in package.json. @haywirebanana  Would it be possible to revert the change made to package-lock.json?

lindan4 committed:
Updated README.md

What did we do?

Update documentation by replacing npm commands with pnpm commands

Why was it done?

Since we are using pnpm, different files will be generated.

lindan4 committed:
Upgraded Astro to 5.1.2

What was done?

Ran pnpm dlx @astrojs/upgrade tp upgrade Astro to the latest version.

why was this done?

To keep Astro up to date. This is major version upgrade, and considering how early we are in the development of the project, it is better to do it now.

haywirebanana committed:
removed package-lock.json -- using pnpm-lock.yaml

haywirebanana committed:
removed package-lock.json -- using pnpm-lock.yaml

lindan4 committed:
Merge pull request #27 from codetocloudinc/feature/nav-bar

Code to Cloud nav bar

lindan4 committed:
Merge pull request #30 from codetocloudinc/add-header

add hero component

lindan4 commented:
<img width="454" alt="Screenshot 2025-02-07 at 7 12 59 PM" src="https://github.com/user-attachments/assets/3f1187fb-cf46-4200-8426-fb4ba1fe220f" />

<img width="844" alt="Screenshot 2025-02-07 at 7 13 21 PM" src="https://github.com/user-attachments/assets/b837b593-90d4-4349-bd59-76f09704bad0" />


@haywirebanana according the mockup we have, it looks like your designs may have to be changed. These include, but are not limited to:

1. There should be no expand/collapse icon (disclosure triangles) on the left and at the bottom, but they should be on the right centred vertically.

2. Font styling and text alignment needs to be changed.

lindan4 commented on src/components/Dropdown.astro:
```diff
@@ -0,0 +1,105 @@
+const { speaker, time, title, body } = Astro.props;
+---
+
+<details class="accordion">
+	<summary class="accordion-header">
+		<div class="header-content">
+			<span class="time">{time}</span>
+			<span class="title">{title}</span>
+            <i class="fa fa-caret-down toggle-icon"></i>
+		</div>
+	</summary>
+	<div class="accordion-body">
+		<p class="body-text">{body}</p>
+		<p class="speaker">{speaker ? `Speaker: ${speaker}` : ''}</p>
+	</div>
+</details>
```
I think we will have to use another outer element to match designs. details would be a lot more convenient though

haywirebanana replied:
With our previous discussions, I needed to visualize what it would look like compared to the old "version" in Figma where it was just plain text. I created that visual on Figma on a whim. Should I adhere to it?

I figured details/summary would be convenient as we are working on a static website, is there an alternative I can use?

lindan4 commented on src/components/Dropdown.astro:
```diff
@@ -0,0 +1,105 @@
+	font-size: 25px;
+}
+
+.body-text {
```
How about accordion-body-text?

lindan4 commented on src/layouts/Layout.astro:
```diff
@@ -14,6 +14,7 @@ const { title } = Astro.props;
 		<meta name="viewport" content="width=device-width" />
 		<link rel="icon" type="image/svg+xml" href="/favicon.svg" />
 		<meta name="generator" content={Astro.generator} />
+    <link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.4/css/all.min.css" rel="stylesheet" />
```
Why do we need this?

haywirebanana replied:
Refer to comments on feature/footer branch

lindan4 commented on src/components/Dropdown.astro:
```diff
@@ -0,0 +1,105 @@
+---
+interface Props {
+	title: string;
+	time: string;
+	body: string;
+	speaker: string;
+}
+
+const { speaker, time, title, body } = Astro.props;
+---
```
I love this

lindan4 reviewed (changes requested):
You'll have to make changes, but also rebase your branch as there are merge conflicts.

haywirebanana committed:
Merge branch 'feature/agenda' of https://github.com/codetocloudinc/code-to-cloud-home into feature/agenda

haywirebanana committed:
rename class name for accordian-body-text
